compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -3262,7 +3262,7 @@ impl<'a> Parser<'a> {
                     id: DUMMY_NODE_ID,
                     is_placeholder: false,
                 },
-                TrailingToken::None,
+                false,
```

Comment:
I would expect a ` next_token == token::Comma` here, because match arms are separated by commas.

I suspect this may cause issues with configured out arms inside derive/cfg_eval.
```rust
#[cfg_eval]
#[proc_macro_that_tries_to_parse_it_again] // will fail with a syntax error because the comma
                                           // is not removed from the token representation?
fn foo() {
  match 10 {
    #[cfg(FALSE)]
    10 => true,
     _ => false,
  }
}
```